core/src/main/scala/org/http4s/Headers.scala

```diff
@@ -87,6 +87,14 @@ final class Headers private (headers: List[Header])
     }
     else super.++(that)
   }
+
+  override def hashCode(): Int = this.headers.hashCode()
+
+  override def equals(that: Any): Boolean = that match {
+    case otherheaders: Headers =>
+      this.hashCode() == otherheaders.hashCode()
```

Comment:
This will be equal in the rare event of a hash collision. Can we just do `this.headers == that.headers`?